File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/BALANCA/Commands/AbrirSerial.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.BALANCA.Commands

import com.elgin.intent_digital_hub.IntentDigitalHubService.BALANCA.BalancaCommand

class AbrirSerial(val baudrate: Int, val lenght: Int, val parity: Char, val stopbits: Int) :
    BalancaCommand<Int>("AbrirSerial") {
    override fun functionParameters(): String {
        return "\"baudrate\"" + ":" + this.baudrate + "," +
                "\"lenght\"" + ":" + this.lenght + "," +
                "\"parity\"" + ":" + "\"" + this.parity + "\"" + "," +
                "\"stopbits\"" + ":" + this.stopbits
    }
}

File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/BALANCA/abrir_serial.dart
import 'package:intent_digital_hub/IntentDigitalHubService/BALANCA/balanca_command.dart';

class AbrirSerial extends BalancaCommand {
  final int _baudrate;
  final int _lenght;
  final String _parity; //parity é um 'char'
  final int _stopbits;

  AbrirSerial(this._baudrate, this._lenght, this._parity, this._stopbits)
      : super('AbrirSerial');

  @override
  String functionParameters() {
    return '"baudrate"'
        ':'
        '$_baudrate'
        ','
        '"lenght"'
        ':'
        '$_lenght'
        ','
        '"parity"'
        ':'
        '"$_parity"'
        ','
        '"stopbits"'
        ':'
        '$_stopbits';
  }
}
